src/main/java/dev/snowdrop/release/services/AutoUpdateService.java

```diff
@@ -0,0 +1,165 @@
+            return null;
+        final StringBuilder ret = new StringBuilder(init.length());
+        for (final String word : init.split(GAV_CAMEL_CASE_REGEX_PATTERN)) {
+            if (!word.isEmpty()) {
+                ret.append(Character.toUpperCase(word.charAt(0)));
+                ret.append(word.substring(1).toLowerCase());
+            }
+        }
+        return ret.toString();
+    }
+
+    /**
+     * <p>Main process for updating the Build Configuration file.</p>
+     *
+     * @param repo
+     * @param release        <p>Release object</p>
+     * @param releaseVersion <p>Release version number</p>
+     * @param qualifier
+     * @param milestone
+     * @return
+     * @throws Throwable
+     */
+    public File updateBuildConfig(File repo, Release release, final String releaseVersion, final String qualifier, final String milestone) {
+        try {
+            final String[] releaseMMFA = releaseVersion.split("\\.");
+            File buildConfigFile = factory.getBuildConfigRelativeTo(repo, releaseVersion);
+            Map<String, String> variableMap = readVariablesFromFile(new FileInputStream(buildConfigFile));
+            variableMap.put("majorVersion", releaseMMFA[0]);
+            variableMap.put("minorVersion", releaseMMFA[1]);
+            variableMap.put("patchVersion", releaseMMFA[2]);
+            variableMap.put("qualifier", qualifier);
+            variableMap.put("milestone", milestone);
+            BuildConfig buildConfigObj = factory.createFromRepo(buildConfigFile);
+            final String[] releaseMMF = releaseVersion.split("\\.");
+            final String releaseMM = releaseMMF[0] + "." + releaseMMF[1];
+            release.getComponents().stream().forEach(component -> {
+//            // TODO: Check if it's a product or a component only template
+//            manageComponentOnly(component);
+                manageProduct(component, variableMap);
+            });
+
+            factory.saveTo(buildConfigObj, generateVariableConfiguration(variableMap), buildConfigFile);
+            return buildConfigFile;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String generateVariableConfiguration(Map<String, String> variableMap) {
+        return variableMap.entrySet()
+            .stream()
+            .map(entry -> "#!" + entry.getKey() + "=" + entry.getValue())
+            .collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    private void manageProduct(Component component, Map<String, String> variableMap) {
+        final Issue issue = issueSvc.getIssue(component);
+        final String description = issue.getDescription();
+        final String versionSection = description.substring(description.lastIndexOf("==="));
+        final String[] artifactArr = versionSection.split("\\n[\\n \\r]*");
+        // Skip the 1st 2 lines
+        for (int arrayPos = 0; arrayPos < ((artifactArr.length - 1) / 5); arrayPos++) {
```

Comment:
Please use constants instead of hardcoded numbers. What does that `5` mean?

Reply:
DONE